lib/iobroker.c

```diff
@@ -140,6 +140,10 @@ struct iobroker_set *iobroker_create(void)
 	}
 
 	iobs->max_fds = iobroker_max_usable_fds();
+	/* add sane max limit, if ulimit is set to unlimited or a very high value we
+	 * don't want to waste memory for nothing */
+	if (iobs->max_fds > 100000)
+		iobs->max_fds = 100000;
```

Comment:
Could we make this configurable just in case someone wants a different value?

Reply:
i thought about it as well. It is usually limited by `ulimit -n`. So if you want a smaller value, set a sane ulimit.
Not sure if a config option is required, maybe a compile time constant?

Reply:
Yeah I guess it is mainly for larger values on would nee to change it. 

There shouldn't be much need to increase it above this limit I suppose, but a configuration option somewhere (compiler/runtime) seems nicer than a "magic" constant.